### PR TITLE
#20154 focal point fix

### LINKED ISSUE #20154: Focal Point is lost when hitting save on the image editor and then publish.
**Describe the bug**
Focal point MD isn't set from content version to version 

**To Reproduce**
1. Let's start by uploading a brand new Image File-Asset
2. Click on Publish
3. Then Open-Up the image editor by double-clicking on the image. Set a Focal Point anywhere.
4. Hit Save
5. At the Content Editor level - Hit Publish.
6. Close and re-open the  Content Editor dialog and verify the Image doesn't have any Focal-point

**Expected behavior**
The Focal Point data must be persisted from version to version after hitting publish. And now it isn't.
Additionally to the compress drop-down isn't working on the Image Editor. And that also needs to be fixed.

### LINKED ISSUE #20192: File Metadata does not get generated if custom attributes are initially written.
**Description**
The real useful File Metadata doesn't get generated if the binary already has custom attributes
This is because the basic heuristic used to generate the metadata says. if it already has metadata do not generate it. 
And the finder method does not differentiate if the metadata present on the binary is made up exclusively from custom attributes. it only states that it already has metadata associated. Therefore the other metadata is never generated.
 
Additionally, when copying metadata from version to version we only must copy custom attributes. if it does make sense. 
I've seen cases in which a binary gets replaced from an image to pdf or txt. and stuff like that.

**To Reproduce**
1. Upload an image, then before publishing it. Open the Image Editor and add a focal point 
2. Hit Save and then Publish the close the window.
3. Now Re-open the Content editor for the same file and verify the MD
4. See There is no MD except for the focal point

If you initially publish the file and then come back to add the focal point it'll work as expected.

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This fixes quite a few issues related with the storage and retrieval of the focal point metadata.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/storage/FileMetadataAPIImpl.java [hunk 1]
```diff
@@ -654,7 +653,7 @@ public void putCustomMetadataAttributes(final String tempResourceId,
                         .storageKey(
                                 new StorageKey.Builder().group(metadataBucketName)
                                         .path(tempResourcePath(tempResourceId))
-                                        .storage(storageType).build())
+                                        .storage(StorageType.FILE_SYSTEM).build())
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/storage/FileMetadataAPIImpl.java:656]: why this?

File: dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java [hunk 2]
```diff
@@ -674,15 +677,34 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
...
-			  fileMetadataAPI.putCustomMetadataAttributes(temp.id, ImmutableMap
-					  .of(fieldVarName, metadata.get().getCustomMeta()));
-		   }
+		if (UtilMethods.isSet(fieldVarName) && UtilMethods.isSet(uuid)) {
+			try {
+				if (UUIDUtil.isUUID(uuid)) {
+					final Contentlet content = contentAPI
+							.find(uuid, APILocator.systemUser(), false);
+					final Metadata metadata = fileMetadataAPI.getMetadata(content, fieldVarName);
+					fileMetadataAPI.putCustomMetadataAttributes(temp.id, ImmutableMap
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java:696]: not sure if it is a good question but; why temp.id is the key

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/storage/FileMetadataAPIImpl.java (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotcms/storage/FileMetadataAPIImpl.java:658]: Currently, we don't have a way to clean up all the temporary stuff that would be created if the storage of choice was the db. But for temp files we do. So for now I think it makes sense if we use the file system to store temp files while we figure out a cleanup mechanism.

File: dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java (hunk 2 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java]: The image editor frequently uses temp files 
Like this http://localhost:8080/contentAsset/image/temp_75f354d10e/fileAsset
in that scenario, we wouldn't have a valid inode So we must work with a temp id

File: dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java [hunk 3]
```diff
@@ -674,15 +678,41 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
...
-		if(UtilMethods.isSet(fieldVarName) && UtilMethods.isSet(uuid)){
-		   final Optional<Metadata> metadata = fileMetadataAPI.getMetadata(uuid);
-		   if(metadata.isPresent()){
-			  fileMetadataAPI.putCustomMetadataAttributes(temp.id, ImmutableMap
-					  .of(fieldVarName, metadata.get().getCustomMeta()));
-		   }
+		if (UtilMethods.isSet(fieldVarName) && UtilMethods.isSet(id)) {
+			try {
+				if (UUIDUtil.isUUID(id)) {
+                   /*
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java:694]: remove commented code

File: dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java [hunk 4]
```diff
@@ -674,15 +678,41 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp) throws Servl
...
+					final Contentlet content = contentAPI
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/servlets/BinaryExporterServlet.java:695]: I think all this code can be reduced to: 
```
final Optional<Metadata> metadata = fileMetadataAPI.getMetadata(UUIDUtil.isUUID(id)? TMP + id:id);
					if (metadata.isPresent()) {
						fileMetadataAPI.putCustomMetadataAttributes(temp.id, ImmutableMap
								.of(fieldVarName, metadata.get().getCustomMeta()));
					}
```

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js [hunk 5]
```diff
@@ -369,22 +356,22 @@ dojo.declare("dotcms.dijit.image.ImageEditor", dijit._Widget,{
...
-            this._removeFocalPoint;
+            this._removeFocalPoint();
             return;
         }
 
 
 
         let ifrm =  window.frames['imageToolIframe']
-        console.log("_drawFocalPoint:" + x + "," + y)
+        //console.log("_drawFocalPoint:" + x + "," + y)
```
Comment (nollymar) [inline dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js:372]: remove the commented code if not needed

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js [hunk 6]
```diff
@@ -369,22 +356,22 @@ dojo.declare("dotcms.dijit.image.ImageEditor", dijit._Widget,{
...
         let coord = dojo.coords(ifrm.contentDocument.getElementById("me"));
-        console.log("_drawFocalPoint:coord:",coord)
+        //console.log("_drawFocalPoint:coord:",coord)
```
Comment (nollymar) [inline dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js:374]: same here

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js [hunk 7]
```diff
@@ -420,10 +407,10 @@ dojo.declare("dotcms.dijit.image.ImageEditor", dijit._Widget,{
     },
 
     _removeFocalPoint: function(){
-        console.log("_removeFocalPoint:")
+        //console.log("_removeFocalPoint:");
```
Comment (nollymar) [inline dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js:410]: same here

File: dotCMS/src/main/java/com/dotmarketing/image/focalpoint/FocalPointAPIImpl.java [hunk 8]
```diff
@@ -57,12 +60,14 @@ public void writeFocalPoint(final String inode, final String fieldVar, final Foc
           return;
         }
 
-        if(tempFileAPI.isTempResource(inode)){
+        if(tempFileAPI.isTempResource(inode) || isSet(inode) && inode.startsWith(TMP)) {
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/image/focalpoint/FocalPointAPIImpl.java:63]: Is it possible to create an IT over this validation?

File: dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js (hunk 7 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js:410]: done.

File: dotCMS/src/main/java/com/dotmarketing/image/focalpoint/FocalPointAPIImpl.java (hunk 8 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/java/com/dotmarketing/image/focalpoint/FocalPointAPIImpl.java]: done.

**Review by nollymar — APPROVED** [review]

**Pull request merged**